lib/arux_app/api.rb

```diff
@@ -1,29 +1,27 @@
 module AruxApp
   module API
-    class << self
-      @@mode = :standard
-      [:test, :dev, :standard].each do |m|
-        define_method("#{m}mode?") do
-          @@mode == m
-        end
+    DOMAINS = {
+      production: "arux.app",
+      staging: "arux.blue",
+      development: HOSTNAME,
+      test: "arux.test"
+    }
 
-        define_method("#{m}mode") do
-          @@mode == m
-        end
+    class << self
+      def mode
+        raise "Environment is not set, i.e. ARUX_APP_GEM_MODE = :development" unless const_defined?(:ARUX_APP_GEM_MODE)
+        ARUX_APP_GEM_MODE
+      end
 
-        define_method("#{m}mode=") do |b|
-          @@mode = b ? m : :standard
-        end
+      def uri(subdomain:)
+        URI::HTTPS.build(
+          host: [subdomain, domain].join('.'),
+        )
       end
 
-      def server_uri
-        if AruxApp::API.standardmode?
-          "https://account.arux.app"
-        elsif AruxApp::API.testmode?
-          "https://account.arux.blue"
-        elsif AruxApp::API.devmode?
-          "https://account.#{HOSTNAME}"
-        end
+      def domain
+        raise "#{mode} is not a supported environment" unless DOMAINS.has_key?(mode)
```

Comment:
Don't need this if we put it in `mode`
```suggestion
```

Reply:
Could be wrong, but I think we need both of them, based on the comment above. Happy to explore more and pair through it if you'd like.